pkg/federate/kubefed/resource_test.go

```diff
@@ -296,13 +298,63 @@ func getResourceFromAPI(kubeFedClient client.Client, resource *unstructured.Unst
 		Kind:    resource.GroupVersionKind().Kind,
 		Version: resource.GroupVersionKind().Version,
 	})
-	_ = kubeFedClient.Get(
+	err := kubeFedClient.Get(
 		context.TODO(),
 		client.ObjectKey{
 			Namespace: resource.GetNamespace(),
 			Name:      resource.GetName(),
 		},
 		fedPod,
 	)
+
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	}
+
 	return fedPod
 }
+
+func testDelete() {
+	var (
+		resource    *corev1.Pod
+		fedResource *unstructured.Unstructured
+		f           *federator
+		err         error
+	)
+
+	BeforeEach(func() {
+		f = &federator{
+			scheme: runtime.NewScheme(),
+		}
+
+		corev1.AddToScheme(f.scheme)
+		resource = newPod("test-pod")
+		fedResource, err = f.createFederatedResource(resource)
+		Expect(err).ToNot(HaveOccurred())
+		f.kubeFedClient = fake.NewFakeClientWithScheme(f.scheme, fedResource)
+	})
+
+	JustBeforeEach(func() {
+		err = f.Delete(resource)
+	})
+
+	When("The resource is succesfully deleted", func() {
```

Comment:
nit: the resulting test output reads a  bit better if the `When` etc text is not capitalized as it prepends _when_.